File: AppGenerateFiles/helpdesk/FrontModel/Tbl_Servicios.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools } from "../../WDevCore/WModules/WComponentsTools.js";
//@ts-ignore
import { ModelProperty } from "../../WDevCore/WModules/CommonModel.js";
import { Cat_Dependencias }  from './Cat_Dependencias.js'
import { Cat_Tipo_Servicio }  from './Cat_Tipo_Servicio.js'
import { CaseTable_Case }  from './CaseTable_Case.js'
import { Tbl_Servicios_Profile }  from './Tbl_Servicios_Profile.js'
class Tbl_Servicios extends EntityClass {
   constructor(props) {
       super(props, 'EntityHelpdesk');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_Servicio;
   /**@type {String}*/ Nombre_Servicio;
   /**@type {String}*/ Descripcion_Servicio;
   /**@type {String}*/ Visibilidad;
   /**@type {String}*/ Estado;
   /**@type {Date}*/ Fecha_Inicio;
   /**@type {Date}*/ Fecha_Finalizacion;
   /**@type {Cat_Dependencias} ManyToOne*/ Cat_Dependencias;
   /**@type {Cat_Tipo_Servicio} ManyToOne*/ Cat_Tipo_Servicio;
   /**@type {Array<CaseTable_Case>} OneToMany*/ CaseTable_Case;
   /**@type {Array<Tbl_Servicios_Profile>} OneToMany*/ Tbl_Servicios_Profile;
}
export { Tbl_Servicios }

File: AppGenerateFiles/helpdesk/Model/Tbl_Servicios.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Tbl_Servicios : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_Servicio { get; set; }
       public string? Nombre_Servicio { get; set; }
       public string? Descripcion_Servicio { get; set; }
       public int? Id_Tipo_Servicio { get; set; }
       public string? Visibilidad { get; set; }
       public string? Estado { get; set; }
       public DateTime? Fecha_Inicio { get; set; }
       public DateTime? Fecha_Finalizacion { get; set; }
       public int? Id_Dependencia { get; set; }
       [ManyToOne(TableName = "Cat_Dependencias", KeyColumn = "Id_Dependencia", ForeignKeyColumn = "Id_Dependencia")]
       public Cat_Dependencias? Cat_Dependencias { get; set; }
       [ManyToOne(TableName = "Cat_Tipo_Servicio", KeyColumn = "Id_Tipo_Servicio", ForeignKeyColumn = "Id_Tipo_Servicio")]
       public Cat_Tipo_Servicio? Cat_Tipo_Servicio { get; set; }
       [OneToMany(TableName = "CaseTable_Case", KeyColumn = "Id_Servicio", ForeignKeyColumn = "Id_Servicio")]
       public List<CaseTable_Case>? CaseTable_Case { get; set; }
       [OneToMany(TableName = "Tbl_Servicios_Profile", KeyColumn = "Id_Servicio", ForeignKeyColumn = "Id_Servicio")]
       public List<Tbl_Servicios_Profile>? Tbl_Servicios_Profile { get; set; }
   }
}
